test/servinge2e/servicemesh_test.go

```diff
@@ -156,92 +160,96 @@ func TestKsvcWithServiceMeshSidecar(t *testing.T) {
 	for _, scenario := range tests {
 		scenario := scenario
 		t.Run(scenario.name, func(t *testing.T) {
-			// Create a ksvc with the specified annotations and labels
-			service := test.Service(scenario.name, serviceMeshTestNamespaceName, serviceMeshTestImage, scenario.annotations)
-			service.ObjectMeta.Labels = scenario.labels
-			service, err := caCtx.Clients.Serving.ServingV1().Services(serviceMeshTestNamespaceName).Create(context.Background(), service, meta.CreateOptions{})
-			if err != nil {
-				t.Errorf("error creating ksvc: %v", err)
-				return
-			}
+			testServiceToService(t, caCtx, serviceMeshTestNamespaceName, scenario)
+		})
+	}
+}
 
-			// Let the ksvc be deleted after test
-			caCtx.AddToCleanup(func() error {
-				t.Logf("Cleaning up Knative Service '%s/%s'", service.Namespace, service.Name)
-				return caCtx.Clients.Serving.ServingV1().Services(serviceMeshTestNamespaceName).Delete(context.Background(), service.Name, meta.DeleteOptions{})
-			})
+func testServiceToService(t *testing.T, ctx *test.Context, namespace string, tc testCase) {
```

Comment:
This func should move to `service_to_service_test.go` or some shared place? I think somebody feels strange that `TestServiceToServiceCalls` calls the tests written in `servicemesh_test.go`.

Reply:
Yeah. Right. Moving to service_to_service_test.go sounds good to me.